File: typescript-examples/auth-with-secret-otp/auth-sessions/check.ts
import { BrowserContext, Page } from "playwright";
import { goToUrl } from "@intuned/browser";

export default async function check(
  page: Page,
  context: BrowserContext
): Promise<boolean> {
  // Step 1: Navigate to a protected page (dashboard)
  await goToUrl({
    page,
    url: "https://sandbox.intuned.dev",
  });

  // Step 2: Check if the user menu toggle is visible
  // The user menu toggle should only be visible on the dashboard if we're logged in
  // If we were redirected to login, this element won't exist
  const userMenuToggle = page.locator("#user-menu-toggle");
  const isSessionValid = await userMenuToggle.isVisible();

  // Return true if the session is still valid (user menu is visible)
  // Return false if the session expired (we were redirected to login)
  return isSessionValid;
}

File: python-examples/starter-auth/auth-sessions/check.py
from intuned_browser import go_to_url
from playwright.async_api import Page


async def check(page: Page, **_kwargs) -> bool:
    # Step 1: Navigate to a protected page (dashboard)
    await go_to_url(
        page=page,
        url="https://sandbox.intuned.dev",
    )

    # Step 2: Check if the user menu toggle is visible
    # The user menu toggle should only be visible on the dashboard if we're logged in
    # If we were redirected to login, this element won't exist
    user_menu_toggle = page.locator("#user-menu-toggle")
    is_session_valid = await user_menu_toggle.is_visible()

    # Return True if the session is still valid (user menu is visible)
    # Return False if the session expired (we were redirected to login)
    return is_session_valid
